src/main/java/org/jitsi/xmpp/component/ComponentBase.java

```diff
@@ -271,10 +306,134 @@ public boolean isConnectionAlive()
+     */
+    @Override
+    final protected IQ handleIQGet(IQ iq)
+        throws Exception
+    {
+        processingTimer.processingStarted(iq);
+        try
+        {
+            return handleIQGetImpl(iq);
+        }
+        finally
+        {
+            processingTimer.processingEnded();
+        }
+    }
+
+    /**
+     * Handles an <tt>IQ</tt> stanza of type <tt>get</tt>.
+     *
+     * @param iq the <tt>IQ</tt> stanza of type <tt>get</tt> which represents
+     * the request to handle.
+     *
+     * @return an <tt>IQ</tt> stanza which represents the response to
+     * the specified request or <tt>null</tt> to reply with
+     * <tt>feature-not-implemented</tt>.
+     *
+     * @throws Exception to reply with <tt>internal-server-error</tt> to the
+     * specified request.
+     *
+     * @see AbstractComponent#handleIQGet(IQ)
+     */
+    protected IQ handleIQGetImpl(IQ iq) throws Exception
+    {
+        return super.handleIQGet(iq);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Should *NEVER* be called by the subclass directly.
+     *
+     * The method is final in order to enforce measurement of the IQ processing
+     * time. Subclasses should implement the processing in
+     * {@link #handleIQSetImpl(IQ)}. The semantics of that method remain
+     * the same as of {@link AbstractComponent#handleIQSet(IQ)}.
+     */
     @Override
-    protected void handleIQResult(IQ iq)
+    final protected IQ handleIQSet(IQ iq)
+        throws Exception
+    {
+        processingTimer.processingStarted(iq);
+        try
+        {
+            return handleIQSetImpl(iq);
+        }
+        finally
+        {
+            processingTimer.processingEnded();
+        }
+    }
+
+    /**
+     * Handles an <tt>IQ</tt> stanza of type <tt>set</tt>.
+     *
+     * @param iq the <tt>IQ</tt> stanza of type <tt>set</tt> which represents
+     * the request to handle.
+     *
+     * @return an <tt>IQ</tt> stanza which represents the response to
+     * the specified request or <tt>null</tt> to reply with
+     * <tt>feature-not-implemented</tt>.
+     *
+     * @throws Exception to reply with <tt>internal-server-error</tt> to the
+     * specified request.
+     *
+     * @see AbstractComponent#handleIQSet(IQ)
+     */
+    protected IQ handleIQSetImpl(IQ iq) throws Exception
+    {
+        return super.handleIQSet(iq);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     *  Should *NEVER* be called by the subclass directly.
+     *
+     * The method is final in order to enforce measurement of the IQ processing
+     * time. Subclasses should implement the processing in
+     * {@link #handleIQResultImpl(IQ)}. The semantics of that method remain
+     * the same as of {@link AbstractComponent#handleIQResult(IQ)}.
+     */
+    @Override
+    final protected void handleIQResult(IQ iq)
+    {
+        try
+        {
+            processingTimer.processingStarted(iq);
```

Comment:
I decided to put inside just because it looks and feels better:
https://github.com/jitsi/jicoco/pull/20/commits/e89c48e11dbe76fc70004f5558ba77c80c202755